src/test/java/study/StringTest.java

```diff
@@ -0,0 +1,60 @@
+package study;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.junit.jupiter.api.Assertions.assertAll;
+
+class StringTest {
+
+    @Test
+    @DisplayName("split() 메소드를 테스트한다.")
+    void split() {
+        // given
+        String data = "1,2";
+
+        // when
+        String[] split = data.split(",");
+
+        // then
+        assertAll(
+                () -> assertThat(split).contains("1"),
+                () -> assertThat(split).contains("2"),
+                () -> assertThat(split).containsExactly("1", "2")
+        );
+    }
+
+    @Test
+    @DisplayName("substring() 메서드를 테스트한다.")
+    void substring() {
+        // given
+        String data = "(1,2)";
+
+        // when
+        String substring = data.substring(1, data.length() - 1);
+
+        // then
+        assertThat(substring).isEqualTo("1,2");
+    }
+
+    @Test
+    @DisplayName("charAt() 메서드를 테스트한다.")
+    void charAt() {
+        // given
+        String data = "abc";
+
+        // when
+
+        // then
+        assertAll(
+                () -> assertThat(data.charAt(0)).isEqualTo('a'),
+                () -> assertThat(data.charAt(1)).isEqualTo('b'),
+                () -> assertThat(data.charAt(2)).isEqualTo('c'),
+                () -> assertThatThrownBy(() -> data.charAt(3))
+                        .isInstanceOf(IndexOutOfBoundsException.class)
+                        .hasMessageContaining("String index out of range: 3")
+        );
```

Comment:
```suggestion
        String data = "abc";

        assertAll(
                () -> assertThat(data.charAt(0)).isEqualTo('a'),
                () -> assertThat(data.charAt(1)).isEqualTo('b'),
                () -> assertThat(data.charAt(2)).isEqualTo('c'),
                () -> assertThatThrownBy(() -> data.charAt(3))
                        .isInstanceOf(IndexOutOfBoundsException.class)
                        .hasMessageContaining("String index out of range: 3")
        );
```

`given, when, then` 구조화를 하면 테스트 코드 작성의 일관성, 구조화 되어 가독성 향상, 간결함 의 장점이 있을거라고 생각되는데요.

그렇다면, 테스트 코드를 간결하게 구현하는게 우선이 아닐까? 라는 생각이 들더라구요.
그런의미에서 공백으로 구분 처리하면 어떨까요?
지금처럼 `//when` 이 없는경우는 굉장이 애매하다고 생각하지 않으셨나요? 🥲

Reply:
지금처럼 의미 없이 //when을 쓰는 경우가 되돌아보니 정말 많았네요! 좋은 리뷰 감사합니다.